syft/pkg/type_test.go

```diff
@@ -119,6 +119,10 @@ func TestTypeFromPURL(t *testing.T) {
 			purl:     "pkg:opam/ocaml-base-compiler@5.2.0",
 			expected: OpamPkg,
 		},
+		{
+			purl:     "pkg:terraform/registry.terraform.io/hashicorp/aws@5.72.1",
+			expected: TerraformPkg,
+		},
```

Comment:
One inconsistency is that in the purl types documentation this purl type is [reserved for modules](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#other-candidate-types-to-define), not providers:

> `terraform` for Terraform modules

Reply:
it might be that we include the cataloger with purls not populated for now until we have a better answer here... this this can at least merge with a follow up when we can get clarification about the purl type

Reply:
I see, yes. Do you want me to remove the purl types?

Reply:
> I see, yes. Do you want me to remove the purl types?

I think it makes sense to remove PURLs from these terraform packages for now. We don't want to have a breaking change if/when purl spec decides to distinguish terraform providers from terraform modules, and PURLs are easy to add back in a follow up if we do get a decision from purl spec.

So I think keeping the _package_ type terraform but dropping the new PURL type and emitting packages with empty PURLs is the right approach for now.

I'll open an issue with the PURL Spec repo to try to clarify the position of terraform modules vs providers, but we don't need to block this PR on that issue.

Please let me know if there's anything I can do to help!

Reply:
I've asked the PURL Spec folks at https://github.com/package-url/purl-spec/issues/369. But as I said, we don't need to block on this issue.